### PR TITLE
Feature/93 azure pipelines cd

### PR DESCRIPTION
**Pull request by alejandro-castellon**
[AB#93](https://dev.azure.com/funjaladev/7867a1c5-5ac7-4c7f-baff-87d740bf4fc1/_workitems/edit/93)

### CODE REVIEW

**Comment by pedro-lima-jalafund** [comment]
please follow [this](https://github.com/fundev3/teamup-webapp/blob/develop/.github/PULL_REQUEST_TEMPLATE.md) for the description of this PR

File: azure-pipelines.yml [hunk 1]
```diff
@@ -14,28 +14,67 @@ pr:
...
-- task: PublishBuildArtifacts@1
-  displayName: 'Dotnet Drop Artifacts'
-  condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
-  inputs:
-    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
-    ArtifactName: 'Drop'
-    publishLocation: 'Container'
+stages:
+- stage: Build
+  displayName: Build stage
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: just `Build`

File: azure-pipelines.yml [hunk 2]
```diff
@@ -14,28 +14,67 @@ pr:
...
+
+    steps:
+    - task: UseDotNet@2
+      displayName: 'DotNet Install'
+      inputs:
+        packageType: 'sdk'
+        version: '3.x'
+
+    - task: DotNetCoreCLI@2
+      displayName: 'Dotnet Build'
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:35]: DotNet

File: azure-pipelines.yml [hunk 3]
```diff
@@ -14,28 +14,67 @@ pr:
...
+      inputs:
+        command: 'build'
+        projects: $(projects)
+        arguments: '--output $(Build.ArtifactStagingDirectory)'
+
+    - task: ArchiveFiles@2
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:47]: displayName: 'Archive files'

File: azure-pipelines.yml [hunk 4]
```diff
@@ -14,28 +14,67 @@ pr:
...
+      inputs:
+        rootFolderOrFile: '$(Build.ArtifactStagingDirectory)'
+        includeRootFolder: false
+        archiveType: zip
+        archiveFile: $(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip
+        replaceExistingArchive: true
+
+    - task: PublishBuildArtifacts@1
+      displayName: 'Dotnet Drop Artifacts'
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: Drop Artifact

File: azure-pipelines.yml [hunk 5]
```diff
@@ -14,28 +14,67 @@ pr:
...
+      condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+      inputs:
+        PathtoPublish: '$(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip'
+        ArtifactName: 'drop'
+        publishLocation: 'Container'
+
+- stage: Deploy
+  displayName: Deploy stage
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: just Deploy

**Review by martin-callejas-jalafund — CHANGES_REQUESTED** [review]
Minor obs! Awesome job

**Review by martin-callejas-jalafund — DISMISSED** [review]
get your other two students approvals and we're good to merge

File: azure-pipelines.yml [hunk 6]
```diff
@@ -11,34 +11,77 @@ pr:
...
+- stage: Deploy
+  displayName: Deploy
+  dependsOn: Build
+  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/develop'))
+
+  jobs: 
+  - job: Deploy
+    displayName: Deploy
+    pool: 
+      vmImage: 'ubuntu-latest'
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: let's remove pool here, we only need it at pipeline level https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#pool

File: azure-pipelines.yml [hunk 7]
```diff
@@ -11,34 +11,77 @@ pr:
...
+    
+    steps:    
+    - task: DownloadBuildArtifacts@1
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:74]: add displayName: Download Artifact

File: azure-pipelines.yml [hunk 8]
```diff
@@ -11,34 +11,77 @@ pr:
...
+      inputs:
+        buildType: 'current'
+        downloadType: 'single'
+        itemPattern: 'drop/$(Build.BuildId).zip'
+        downloadPath: '$(System.ArtifactsDirectory)'
+    - task: AzureFunctionApp@1
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:81]: add displayName: Deploy API

**Review by martin-callejas-jalafund — CHANGES_REQUESTED** [review]

**Review by fernando-maceda-jalafund — DISMISSED** [review]
LGTM

**Review by paulo-teran-jalafund — DISMISSED** [review]
LGTM

**Review by pedro-lima-jalafund — DISMISSED** [review]

File: azure-pipelines.yml [hunk 9]
```diff
@@ -11,34 +11,77 @@ pr:
...
+  projects: '**/**/*.sln'
+
+stages:
+- stage: Build
+  displayName: Build
+
+  jobs: 
+  - job: Build
+    pool:
+      vmImage: ubuntu-latest
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml]: move to pipeline level

**Review by martin-callejas-jalafund — CHANGES_REQUESTED** [review]

**Review by martin-callejas-jalafund — APPROVED** [review]

**Review by pedro-lima-jalafund — APPROVED** [review]

**Review by paulo-teran-jalafund — APPROVED** [review]

**Review by rodrigo-baldiviezo-jalafund — APPROVED** [review]

**Pull request merged**